optaweb-tsp-ui/package-lock.json

```diff
@@ -4,6 +4,12 @@
   "lockfileVersion": 1,
```

Comment:
Should package-lock.json files be checked in? Seems like something that should be gitignored (preferably because its in the target directory, but if that's painful, we can add it to gitignore directly).

Reply:
Actually it should be committed because:

> - Describe a single representation of a dependency tree such that teammates, deployments, and continuous integration are guaranteed to install exactly the same dependencies.
> - Provide a facility for users to "time-travel" to previous states of node_modules without having to commit the directory itself.
> - To facilitate greater visibility of tree changes through readable source control diffs.
> - And optimize the installation process by allowing npm to skip repeated metadata resolutions for previously-installed packages.
[npm docs](https://github.com/npm/npm/blob/v5.0.0/doc/files/package-lock.json.md#description)

Otherwise ignoring it allow us to always test the latest version of dependencies (especially in CI).

Reply:
AFAIU pacakge-lock.json is necessary for reproducible builds and it should be version-controlled.